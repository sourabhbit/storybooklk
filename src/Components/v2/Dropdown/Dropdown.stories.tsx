import { ComponentMeta, ComponentStory } from '@storybook/react'
import {
	generateArgTypes,
	generateGeneralArgs,
} from '../../../Helpers/Storybook'
import {
	ComponentSizeENUM,
	ThemeENUM,
	TypographyENUM,
} from '../../../Types/general'
import Dropdown from './index'

export default {
	title: 'Components/v2/Dropdown',
	component: Dropdown,
	argTypes: {
		...generateGeneralArgs(),
		options: generateArgTypes({
			type: 'array',
			required: true,
			description: 'Show dropdown options',
			category: 'Component',
		}),
		index: generateArgTypes({
			type: 'number',
			required: true,
			description: 'Show cuurent active index',
			category: 'Component',
		}),
	},
} as ComponentMeta<typeof Dropdown>

const Template: ComponentStory<typeof Dropdown> = (arges) => (
	<Dropdown {...arges} />
)

export const Overview = Template.bind({})
Overview.args = {
	id: 'dropdrow-id',
	componentSize: ComponentSizeENUM.small,
	theme: ThemeENUM.secondary,
	font: TypographyENUM.serif,
	options: ['Your Picks', 'Discover', 'Categories', 'Gold', 'Collabs', 'Labs'],
	SelectedIndex: 0,
}
