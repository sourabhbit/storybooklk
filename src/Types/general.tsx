export enum ComponentSizeENUM {
	small = 'small',
	medium = 'medium',
	large = 'large',
}

export enum TypographyENUM {
	sans = 'var(--font-sans)',
	serif = 'var(--font-serif)',
}

export enum ThemeENUM {
	primary = 'primary',
	secondary = 'secondary',
}
