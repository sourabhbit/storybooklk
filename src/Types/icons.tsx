export enum iconsENUM {
	addToCart = 'addToCart',
	addedToCart = 'addedToCart',
	antiScratch = 'antiScratch',
	backArrow = 'backArrow',
	bin = 'bin',
	cross = 'cross',
	crossWithSpace = 'crossWithSpace',
	edit = 'edit',
	emptyCart = 'emptyCart',
	fullWidth = 'fullWidth',
	gift = 'gift',
	glasses = 'glasses',
	globe = 'globe',
	gridList = 'gridList',
	phone = 'phone',
	share = 'share',
	star = 'star',
	starFilled = 'starFilled',
	tick = 'tick',
	user = 'user',
	group = 'group',
}
