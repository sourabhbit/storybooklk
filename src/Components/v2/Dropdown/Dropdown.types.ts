import { ComponentSizeENUM, ThemeENUM, TypographyENUM } from 'Types/general'
export interface DropdownTypes {
	id: string
	options: string[]
	SelectedIndex?: number
	theme: ThemeENUM
	font: TypographyENUM
	componentSize?: ComponentSizeENUM
}
