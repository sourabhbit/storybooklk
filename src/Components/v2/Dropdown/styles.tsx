import styled from 'styled-components'
import { ComponentSizeENUM, ThemeENUM, TypographyENUM } from 'Types/general'

const getSize = (size: ComponentSizeENUM) => {
	if (size === 'small') return 'var(--fs-12)'
	if (size === 'medium') return 'var(--fs-16)'
	else return 'var(--fs-18)'
}

const getBGColor = (colorType: ThemeENUM) => {
	if (colorType === ThemeENUM.primary) {
		return 'var(--white)'
	} else {
		return 'var(--dark-blue-100)'
	}
}

const getColor = (colorType: ThemeENUM, count: boolean) => {
	if (colorType === ThemeENUM.primary) {
		return count ? 'var(--dark-blue-50)' : 'var(--dark-blue-100)'
	} else {
		return count ? 'var(--see-green-25)' : 'var(--see-green-100)'
	}
}

const getFontSize = (size: ComponentSizeENUM) => {
	if (size === ComponentSizeENUM.small) {
		return 'var(--fs-12)'
	} else if (size === ComponentSizeENUM.medium) {
		return 'var(--fs-18)'
	} else {
		return 'var(--fs-30)'
	}
}

export const DropdownWrapper = styled.div<{
	styleColor: ThemeENUM
	styleSize: ComponentSizeENUM
	styleFont: TypographyENUM
}>`
	display: flex;
	flex-direction: column;
	color: ${(props) => getColor(props.styleColor, false)};
	font-size: ${(props) => getFontSize(props.styleSize)};
	font-family: ${(props) => props.styleFont};
`

export const SelectWrapper = styled.div<{
	styleColor: ThemeENUM
	styleSize: ComponentSizeENUM
	showOptions: boolean
}>`
	display: flex;
	align-items: center;
	justify-content: center;
	width: 100px;
	height: 40px;
	background-color: ${(props) => getBGColor(props.styleColor)};
	border: none;
	outline: none;
	border-radius: ${(props) =>
		props.showOptions
			? `var(--border-radius-xs) var(--border-radius-xs) 0 0`
			: `var(--border-radius-xs)`};
	text-align: center;
	font-size: ${(props) => getSize(props.styleSize)};
	cursor: pointer;
`

export const OptionWrapper = styled.span<{
	styleSize: ComponentSizeENUM
	styleColor: ThemeENUM
}>`
	display: flex;
	align-items: center;
	justify-content: center;
	background-color: ${(props) => getBGColor(props.styleColor)};
	width: 100px;
	height: 40px;
	border: none;
	text-align: center;
	font-size: ${(props) => getSize(props.styleSize)};
	cursor: pointer;
	&:last-child {
		border-radius: 0 0 var(--border-radius) var(--border-radius);
	}
`
