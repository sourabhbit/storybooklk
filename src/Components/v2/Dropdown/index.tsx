import { DropdownWrapper, SelectWrapper, OptionWrapper } from './styles'
import { DropdownTypes } from './Dropdown.types'
import { ComponentSizeENUM, ThemeENUM, TypographyENUM } from 'Types/general'
import React from 'react'

const Dropdown = ({
	id,
	options,
	SelectedIndex = 0,
	componentSize = ComponentSizeENUM.small,
	theme = ThemeENUM.primary,
	font = TypographyENUM.serif,
}: DropdownTypes) => {
	const [showOptions, setShowOptions] = React.useState(false)
	const [currentIndex, setCurrentIndex] = React.useState(SelectedIndex)

	const onChangeaction = () => {
		setShowOptions(!showOptions)
	}

	const changeIndex = (index: number) => {
		setCurrentIndex(index)
	}

	return (
		<DropdownWrapper
			id={id}
			data-testid="dropdown-wrapper"
			styleColor={theme}
			styleFont={font}
			styleSize={componentSize}
			onClick={() => onChangeaction()}
		>
			<SelectWrapper
				data-testid="select-wrapper"
				showOptions={showOptions}
				styleSize={componentSize}
				styleColor={theme}
			>
				{options[currentIndex]}
			</SelectWrapper>
			{showOptions &&
				options?.map((option, index) => {
					return (
						<OptionWrapper
							data-testid="option-wrapper"
							onClick={() => changeIndex(index)}
							styleSize={componentSize}
							styleColor={theme}
							key={index}
						>
							{option}
						</OptionWrapper>
					)
				})}
		</DropdownWrapper>
	)
}

export default Dropdown
