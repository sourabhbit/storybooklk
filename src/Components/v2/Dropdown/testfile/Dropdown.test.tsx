import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { ComponentSizeENUM, ThemeENUM, TypographyENUM } from 'Types/general'
import Dropdown from '../index'

const props = {
	id: 'dropdrow-id',
	componentSize: ComponentSizeENUM.small,
	theme: ThemeENUM.secondary,
	font: TypographyENUM.serif,
	options: ['Your Picks', 'Discover', 'Categories', 'Gold', 'Collabs', 'Labs'],
	SelectedIndex: 0,
}

describe('<Dropdown />', () => {
	let _asFragment: any = null

	beforeEach(() => {
		const { asFragment } = render(<Dropdown {...props} />)
		_asFragment = asFragment
	})

	afterAll(() => {
		_asFragment = null
	})

	it('Should render component', () => {
		const container = screen.getByTestId('dropdown-wrapper')
		expect(container).toBeInTheDocument()
	})

	it('Should take a snapshot', () => {
		expect(_asFragment()).toMatchSnapshot()
	})
})
