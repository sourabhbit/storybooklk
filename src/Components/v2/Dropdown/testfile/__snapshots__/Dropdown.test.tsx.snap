// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> Should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL jdbxvH"
    data-testid="dropdown-wrapper"
    id="dropdrow-id"
  >
    <div
      class="sc-gsDKAQ eOEBkQ"
      data-testid="select-wrapper"
    >
      Your Picks
    </div>
  </div>
</DocumentFragment>
`;
